src/runtime/HalideBuffer.h

```diff
@@ -1991,9 +1991,11 @@ class Buffer {
     /** Make a buffer with the same shape and memory nesting order as
      * another buffer. It may have a different type. */
     template<typename T2, int D2, int S2>
-    static Buffer<T, Dims, InClassDimStorage> make_with_shape_of(const Buffer<T2, D2, S2> &src,
+    static Buffer<T, Dims, InClassDimStorage> make_with_shape_of(Buffer<T2, D2, S2> src,
```

Comment:
We'll probably need to add a NOLINT comment here once we move to clang-tidy-19 but that can (and should) be done in the other CL since exact placement and syntax may be a bit fussy: https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics